pyproject.toml

```diff
@@ -34,7 +34,8 @@ readme = "README.md"
 # Investigate if it is better to fix specific versions or use lower and upper bounds  
 # FIXME: https://github.com/zama-ai/concrete-ml-internal/issues/2665
 python = ">=3.8.1,<3.11"
-concrete-python = "2.6.0-rc1"
+concrete-python = {version="==2.6.2.dev20240529", source = "zama-pypi"}
+# concrete-python = ">=2.6.2.dev20240523,<3.0"
```

Comment:
Couldn't use that requirement because 2.6.2 > 2.6.2dev, so poetry lock would install 2.6.2

Reply:
but we want to fix CP's version no ?

Reply:
Well that's a good question. I'm not so sure about that.

In theory we should only require the major version of a package unless we know that there is a change that breaks our code base.

Introducing more flexibility here would let us avoid the need of doing a patch release whenever Concrete Python does ones. That wouldn't be reflected in the lock file though

Reply:
yeah for our other libraries I pretty much agree, but CP is a bit different, especially since we consider nightly versions and I don't think we would want poetry to consider the most recent nightly each time we update the lock (?)